### ISSUE TITLE
Auto-cast numbers to strings where needed

### ISSUE THREAD

**Issue by sbliven** [citation-file-format/schema#12]
pykwalifire currently fails CFF files that use bare numbers for String fields. For instance, `number: 1` fails and ironically must be changed to `number: "1"` (as pointed out by @jspaaks in #9). This is unexpected since most languages would automatically cast the integer to string if needed.

Would it be reasonable to accept `str | int` for string fields (assuming pykwalifire supports type unions somehow)?

**Comment by sbliven** [comment]
I just read your comment in #11 about switching to YAtiML. It looks like pykwalifire doesn't support unions but that YAtiML does!

**Comment by sdruskat** [comment]
Yes, luckily. This is one of the reasons why we want to make the switch :).

I'll leave this open for reference when we implement the schema in YAtiML.

**Comment by sdruskat** [comment]
Transferring this to https://github.com/citation-file-format/citation-file-format as part of merging the repositories.
